### PR TITLE
Adding Cert sans to kubeadm config

### PR DESCRIPTION
**Pull request by juliocri**
When we are using instances in the public cloud, such amazon or azure, we need a way to connect to the public k8s cluster with the public ip or the public name, with this addition we are supporting the ability to add those public values into the k8s certificates in order to can connect from anywhere not only from the private cluster subnet.

example: 

```
export CERT_SANS="34.221.115.244 ec2-34-221-115-244.us-west-2.compute.amazonaws.com"

 ./create_stack.sh  init
```

```
$kubectl cluster-info
Kubernetes master is running at https://ec2-34-221-115-244.us-west-2.compute.amazonaws.com:6443
KubeDNS is running at https://ec2-34-221-115-244.us-west-2.compute.amazonaws.com:6443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy
```

### CODE REVIEW

**Comment by krsna1729** [comment]
This is getting ugly. @jascott1 can we migrate to kustomize for kubeadm.yaml?

**Comment by juliocri** [comment]
> This is getting ugly. @jascott1 can we migrate to kustomize for kubeadm.yaml?

@jascott1 how do you see this approach? #193

**Comment by juliocri** [comment]
@jascott1 do you think this addition is worthy adding it as a env var or should be though patches?

**Comment by jascott1** [comment]
@juliocri Im not sure how we can avoid `sed` here. As ugly as its becoming, it is still simple if keep the approach in #193 and use these seds only when necessary.

**Comment by jascott1** [comment]
@juliocri please rebase this and I will test it ASAP. Thanks!

**Comment by juliocri** [comment]
> @juliocri please rebase this and I will test it ASAP. Thanks!

Sure @jascott1, This is now rebased, Thanks!

File: clr-k8s-examples/create_stack.sh [hunk 1]
```diff
@@ -53,7 +54,12 @@ function cluster_init() {
 	if [[ -n "$K8S_VER" && $(grep -c kubernetesVersion ./kubeadm.yaml) -eq 0 ]]; then
 		sed -i "s/ClusterConfiguration/ClusterConfiguration\nkubernetesVersion: ${K8S_VER}/g" ./kubeadm.yaml
 	fi
-	# Config patches
+	if ! [ -z "${CERT_SANS}" ]; then
```
Comment (jascott1) [inline clr-k8s-examples/create_stack.sh]: `! [ -z "${CERT_SANS}" ]` could be `[[ -n "${CERT_SANS}" ]]`

**Comment by jascott1** [comment]
@juliocri Could you please make this idempotent? Something like `[[ $(grep -c "certSANs" kubeadm.yaml) -gt 0 ]] || do sed here`

**Comment by juliocri** [comment]
@jascott1 now cert_sans add all yaml properties if are not set (first time) or only replaces the list if certSANs was added previously.

**Comment by jascott1** [comment]
@juliocri improved but still adds duplicate list values on subsequent re-runs
```
 CERT_SANS='34.221.115.244 ec2-34-221-115-244.us-west-2.compute.amazonaws.com' ./create_stack.sh init
...
$ cat kubeadm.yaml
apiVersion: kubeadm.k8s.io/v1beta1
kind: ClusterConfiguration
apiServer:
  certSANs:
  - ec2-34-221-115-244.us-west-2.compute.amazonaws.com
  - 34.221.115.244
  - ec2-34-221-115-244.us-west-2.compute.amazonaws.com
  - 34.221.115.244
  - ec2-34-221-115-244.us-west-2.compute.amazonaws.com
  - 34.221.115.244
  - ec2-34-221-115-244.us-west-2.compute.amazonaws.com
  - 34.221.115.244
networking:
  dnsDomain: cluster.local
  podSubnet: 10.244.0.0/16
  serviceSubnet: 10.96.0.0/12
```

This is important as we sometimes call init and then join nodes and then call `minimal` or `all` with create_stack.sh.

**Comment by juliocri** [comment]
@jascott1 sorry about it bad regex, now the values are not duplicated.

**Pull request merged**